dask/array/ufunc.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import absolute_import, division, print_function
+
+from operator import getitem
+
+import numpy as np
+
+from toolz.curried import merge
+from .core import Array, elemwise
+from .. import core
+from ..utils import skip_doctest
+
+
+def __array_wrap__(numpy_ufunc, x, *args, **kwargs):
+    return x.__array_wrap__(numpy_ufunc(x, *args, **kwargs))
+
+
+def wrap_elemwise(numpy_ufunc, array_wrap=False):
+    """ Wrap up numpy function into dask.array """
+
+    def wrapped(x, *args, **kwargs):
+        if hasattr(x, '_elemwise'):
+            if array_wrap:
+                return x._elemwise(__array_wrap__, numpy_ufunc,
```

Comment:
is `__array_wrap__` defined for dask.array? Why exactly do you need this, only for some ufuncs?